tests/dialects/test_snowflake.py

```diff
@@ -1882,6 +1882,56 @@ def test_copy(self):
         self.validate_identity(
             """COPY INTO mytable (col1, col2) FROM 's3://mybucket/data/files' FILES = ('file1', 'file2') PATTERN = 'pattern' FILE_FORMAT = (FORMAT_NAME = my_csv_format NULL_IF = ('str1', 'str2')) PARSE_HEADER = TRUE"""
         )
+        self.validate_all(
+            """COPY INTO 's3://example/data.csv'
+    FROM EXTRA.EXAMPLE.TABLE
+    credentials = (x)
+    STORAGE_INTEGRATION = S3_INTEGRATION
+    FILE_FORMAT = (TYPE = CSV COMPRESSION = NONE NULL_IF = ('') FIELD_OPTIONALLY_ENCLOSED_BY = '"')
+    HEADER = TRUE
+    OVERWRITE = TRUE
+    SINGLE = TRUE
+            """,
+            write={
+                "": """COPY INTO 's3://example/data.csv'
+FROM EXTRA.EXAMPLE.TABLE
+CREDENTIALS = (x) WITH (
+  STORAGE_INTEGRATION S3_INTEGRATION,
+  FILE_FORMAT = (TYPE = CSV COMPRESSION = NONE NULL_IF = (
+    ''
+  ) FIELD_OPTIONALLY_ENCLOSED_BY = '"'),
+  HEADER TRUE,
+  OVERWRITE TRUE,
+  SINGLE TRUE
+)""",
```

Comment:
Is there a purpose for testing the transpilation from Snowflake to the base dialect other than pretty print? 

The base dialect converted to T-SQL/DuckDB/Postgres format (which have the 'WITH' syntax), and generally COPY options from one dialect will hardly match to other dialects, so the base dialect query does not conform to any specs

Reply:
the base dialect spec needs to work properly, in main it crashes

Reply:
it crashed